pkg/skaffold/kubernetes/log.go

```diff
@@ -104,7 +104,7 @@ func (a *LogAggregator) Start(ctx context.Context) error {
 					continue
 				}
 
-				for _, container := range pod.Status.ContainerStatuses {
+				for _, container := range append(pod.Status.ContainerStatuses, pod.Status.InitContainerStatuses...) {
```

Comment:
InitContainers run to completion. Running `kubeclt logs --since` for completed initContainer is going to be expensive.
```
tejaldesai@tejaldesai~/initcontainers:$ time kubectl logs --since=20s -f init-demo -c install 

real	0m5.120s
user	0m0.105s
sys	0m0.020s

tejaldesai@tejaldesai~/initcontainers:$ time kubectl logs init-demo -c install 
init is running!
init is up again.

real	0m0.108s
user	0m0.106s
sys	0m0.015s
tejaldesai@tejaldesai~/initcontainers:$
```
this is going to slow down a lot if a user has multiple init containers. 
Please add a check to filter completed init containers if its run to completion.

It would be great to see if you add an integration test like @balopat mentioned.